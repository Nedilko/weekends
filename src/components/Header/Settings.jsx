import { useState } from 'react'
import SettingsIcon from '@components/Header/SettingsIcon'
import SettingsModal from '@UI/Modals/SettingsModal'
import { useSettingsData, useSettingsDispatch } from '@store/Settings'

function Settings() {
  const [isModalOpened, seIsModalOpened] = useState(false)
  const settings = useSettingsData()
  const dispatch = useSettingsDispatch()

  const handleOpen = () => {
    seIsModalOpened(true)
  }
  const handleClose = () => {
    seIsModalOpened(false)
  }
  const handleApply = (data) => {
    dispatch({ type: 'set', payload: data })
    seIsModalOpened(false)
  }

  return (
    <>
      <SettingsIcon onClick={handleOpen} />
      {isModalOpened && (
        <SettingsModal
          title="Settings"
          onApply={handleApply}
          onCancel={handleClose}
          settings={settings}
        />
      )}
    </>
  )
}

export default Settings
