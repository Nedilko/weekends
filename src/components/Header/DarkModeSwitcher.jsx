import { useEffect, useState } from 'react'
import ToggleTheme from '@UI/Toggles/ToggleTheme'
import { useSettingsData, useSettingsDispatch } from '@store/Settings'
import { getSystemTheme, applyTheme } from '@utils/settings'

function DarkModeSwitcher() {
  const settings = useSettingsData()
  const dispatch = useSettingsDispatch()
  const { useSystemTheme, theme } = settings
  const [isDarkMode, setIsDarkMode] = useState(() => {
    if (useSystemTheme) return getSystemTheme() === 'dark'
    return theme === 'dark'
  })

  useEffect(() => {
    const currentTheme = useSystemTheme ? getSystemTheme() : theme
    setIsDarkMode(currentTheme === 'dark')
    applyTheme(currentTheme)
  }, [theme, useSystemTheme])

  return (
    <>
      {!useSystemTheme && (
        <ToggleTheme
          isChecked={isDarkMode}
          onClick={() =>
            dispatch({
              type: 'setTheme',
              payload: isDarkMode ? 'light' : 'dark',
            })
          }
        />
      )}
    </>
  )
}

export default DarkModeSwitcher
