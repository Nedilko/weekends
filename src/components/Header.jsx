import DarkModeSwitcher from './DarkModeSwitcher'

function Header() {
  return (
    <div className="flex absolute right-0 justify-end p-2">
      <DarkModeSwitcher />
    </div>
  )
}

export default Header
