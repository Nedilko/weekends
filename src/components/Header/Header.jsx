import DarkModeSwitcher from './DarkModeSwitcher'
import Settings from './Settings'

function Header() {
  return (
    <header className="flex justify-between bg-white dark:bg-zinc-900 p-2">
      <Settings />
      <DarkModeSwitcher />
    </header>
  )
}

export default Header
