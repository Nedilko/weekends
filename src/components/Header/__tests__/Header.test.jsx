import Header from '@components/Header/Header'
import { render, screen } from '@utils/test-utils'

jest.mock('@components/Header/Settings', () => {
  const originalModule = jest.requireActual('@components/Header/Settings')
  return {
    __esModule: true,
    ...originalModule,
    default: () => {
      return <div>Settings</div>
    },
  }
})

jest.mock('@components/Header/DarkModeSwitcher', () => {
  const originalModule = jest.requireActual(
    '@components/Header/DarkModeSwitcher'
  )
  return {
    __esModule: true,
    ...originalModule,
    default: () => {
      return <div>Switcher</div>
    },
  }
})

describe('Header', () => {
  it('should render setting icon and theme switcher', () => {
    render(<Header />)
    expect(screen.getByText(/settings/i)).toBeInTheDocument()
    expect(screen.getByText(/switcher/i)).toBeInTheDocument()
  })

  it('should render header component', () => {
    render(<Header />)
    expect(screen.getByTestId('header')).toMatchSnapshot()
  })
})
