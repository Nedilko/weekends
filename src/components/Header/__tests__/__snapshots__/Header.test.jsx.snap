// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header component 1`] = `
<header
  class="my-2 mx-4 flex h-12 flex-row justify-between"
  data-testid="header"
>
  <div>
    Settings
  </div>
  <div>
    Switcher
  </div>
</header>
`;
